From 96322a4915582f69c5d7ef1816c215885b681ba7 Mon Sep 17 00:00:00 2001
From: Justin Cinkelj <justin.cinkelj@xlab.si>
Date: Wed, 30 Aug 2017 16:37:06 +0200
Subject: [PATCH] wrk: remove check of file descriptor value

The check incorrectly fails on OSv. With say 1 conection, max expected
fd was 10+3*1. Once other apps consume some fds, the check fails.

Signed-off-by: Justin Cinkelj <justin.cinkelj@xlab.si>
---
 src/ae.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/src/ae.c b/src/ae.c
index 90be4e2..ef8ca01 100644
--- a/src/ae.c
+++ b/src/ae.c
@@ -105,9 +105,14 @@ void aeStop(aeEventLoop *eventLoop) {
 int aeCreateFileEvent(aeEventLoop *eventLoop, int fd, int mask,
         aeFileProc *proc, void *clientData)
 {
+    // The eventLoop->setsize is set by:
+    //  ./src/wrk.c:145:        t->loop        = aeCreateEventLoop(10 + cfg.connections * 3);
+    // This does not work on OSv. FD numbers are shared by all apps,
+    // and can get arbitrary large.
     if (fd >= eventLoop->setsize) {
-        errno = ERANGE;
-        return AE_ERR;
+        //fprintf(stderr, "INFO: wrk, allow large fd=%d > %d in aeCreateFileEvent on OSv\n", fd, eventLoop->setsize);
+        //errno = ERANGE;
+        //return AE_ERR;
     }
     aeFileEvent *fe = &eventLoop->events[fd];
 
-- 
2.9.4

